src/components/search/search.jsx

```diff
@@ -0,0 +1,367 @@
+import {
+	useState,
+	forwardRef,
+	useRef,
+	createContext,
+	useContext,
+	Children,
+} from 'react';
+import { cn } from '@/utilities/functions';
+import { Search } from 'lucide-react';
+import Loader from '../loader';
+import {
+	textSizeClassNames,
+	variantClassNames,
+	disabledClassNames,
+	sizeClassNames,
+	IconClasses,
+} from './styles';
+import {
+	autoUpdate,
+	flip,
+	FloatingPortal,
+	offset,
+	size as floatingSize,
+	useDismiss,
+	useFloating,
+	useInteractions,
+} from '@floating-ui/react';
+
+const SearchContext = createContext();
+
+const useSearchContext = () => {
+	return useContext( SearchContext );
+};
+
+const SearchBox = forwardRef(
+	(
+		{
+			className, // Custom class names for additional styling
+			size = 'sm', // Size of the SearchBox (default is 'sm')
+			open = false, // Controls whether the dropdown is open or not (default is false)
+			onOpenChange = () => {}, // Callback function to handle the open state change
+			...props // Any additional props that are passed
```

Comment:
Maybe you shouldn't leave comments here, since you explain the props in the readme file.